### PR TITLE
fix(dashboards): added error capture for dashboard dereferencing issue

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Added error capturing on `data` when `value` is `undefined` to debug.

### CODE REVIEW

File: static/app/views/eventsV2/table/queryField.tsx [hunk 1]
```diff
@@ -101,18 +102,32 @@ type OptionType = {
...
-      </components.Option>
-    ),
-    SingleValue: ({data, ...props}: SingleValueProps<OptionType>) => (
-      <components.SingleValue data={data} {...props}>
-        <span data-test-id="label">{data.label}</span>
-        {this.renderTag(data.value.kind)}
-      </components.SingleValue>
-    ),
+    Option: ({label, data, ...props}: OptionProps<OptionType>) => {
+      if (data.value === undefined)
```
Comment (mrbaguvix) [inline static/app/views/eventsV2/table/queryField.tsx:106]:
```suggestion
      if (data.value === '' || data.value === undefined)
```

File: static/app/views/eventsV2/table/queryField.tsx [hunk 2]
```diff
@@ -101,18 +102,32 @@ type OptionType = {
...
+        });
+      return (
+        <components.Option label={label} data={data} {...props}>
+          <span data-test-id="label">{label}</span>
+          {this.renderTag(data.value.kind)}
+        </components.Option>
+      );
+    },
+    SingleValue: ({data, ...props}: SingleValueProps<OptionType>) => {
+      if (data.value === undefined)
```
Comment (mrbaguvix) [inline static/app/views/eventsV2/table/queryField.tsx:119]:
```suggestion
      if (data.value === '' || data.value === undefined)
```

**Review by 0ex-d — APPROVED** [review]

File: static/app/views/eventsV2/table/queryField.tsx (hunk 1 above)
Comment (k-fish) [reply static/app/views/eventsV2/table/queryField.tsx:106]: @edwardgou-sentry You're missing some (preferred) braces here, and you can just use truthiness or the `defined` function.

**Review by k-fish — APPROVED** [review]

File: static/app/views/eventsV2/table/queryField.tsx (hunk 1 above)
Comment (mrbaguvix) [reply static/app/views/eventsV2/table/queryField.tsx:106]: Totally agree truthiness is not completely covered here.

**Comment by edwardgou-sentry** [comment]
Updated to using !truthy and added braces!

**Pull request merged**